content/blogs/2025-03-27-using-amazon-s3-tables-with-kestra.md

```diff
@@ -0,0 +1,307 @@
+---
+title: "Using Amazon S3 Tables with Kestra"
+description: "A step-by-step walkthrough of how we can orchestrate data loading into Amazon S3 tables using Kestra."
+date: 2025-03-27T17:00:00
+category: Solutions
+author:
+  name: Shruti Mantri
+  image: smantri
+  role: 
+---
+
+Amazon recently introduced S3 Tables, purpose-built for storing and querying tabular data directly on S3. Backed by built-in Apache Iceberg support, S3 Tables make data instantly accessible to popular AWS and third-party analytics engines like EMR and Athena.
+
+In this guide, we’ll show you how to orchestrate a complete workflow using Kestra—from downloading raw CSV files to converting them, uploading to S3, and creating Iceberg-backed S3 Tables. You’ll also learn how to query the data using Athena.
+
+## Why S3 Tables and Kestra?
+
+Structured data is often stored in files across object storage systems like S3—but making it queryable usually requires manual setup, format conversion, and provisioning of compute engines.
+
+With Kestra, you can automate this entire process. Our declarative workflows handle data conversion, orchestration logic, infrastructure interactions, and job submission to EMR—all in a repeatable and trackable way.
+
+This walkthrough will help you:
+
+- Ingest and convert data into Parquet format
+
+- Upload the data into S3
+
+- Create an Iceberg table backed by S3 Table Buckets
+
+- Automate querying through Athena
+
+## Prerequisites
+
+We will need the following as a prerequisite before we could proceed with the workflow creation:
+
+- Kestra server running on version >= 0.21.0
+- AWS account with access to IAM, S3 and Athena
+
+Ensure you select the AWS region in which S3 Tables are [supported](https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-tables-regions-quotas.html).
+
+## Implementing Kestra Workflow for S3 Tables
+
+In order to get the end-to-end Kestra workflow that interacts with S3 table, we will need a lot of prework to be done. We will go through each of the step in detail. This will involve making some changes on the AWS console. In the process, we will also develop the Kestra workflow incrementally.
+
+### Create general purpose S3 bucket
+
+Firstly, we will need a general purpose S3 bucket where we will store the data. For this, navigate to the [S3 service](https://console.aws.amazon.com/s3/home) on the AWS console. From the left navigation menu, select `General purpose buckets`. On the `General purpose buckets` page, click on the `Create bucket` button. On the `Create bukcet` page, provide a globally unique bucket name in the `Bucket name` text box. For the purpose of this blog, say we put the bucket name as `s3-general-purpose-ecommerce`. Rest of the configurations can be left as default, and click on the **Create bucket** button at the bottom of the page. This will create the new bucket.
```

Comment:
```suggestion
Firstly, we will need a general purpose S3 bucket where we will store the data. For this, navigate to the [S3 service](https://console.aws.amazon.com/s3/home) on the AWS console. From the left navigation menu, select `General purpose buckets`. On the `General purpose buckets` page, select the `Create bucket` button. On the `Create bucket` page, provide a globally unique bucket name in the `Bucket name` text box. For the purpose of this blog, for example we name the bucket as `s3-general-purpose-ecommerce`. Rest of the configurations can be left as default, and select the **Create bucket** button at the bottom of the page. This will create the new bucket.
```